### ISSUE TITLE
🛠️ Repo: Switch automations to GitHub Actions (GHA)

### ISSUE THREAD

**Issue by craigtaub** [mochajs/mocha#4117]
Can we make some of the manual processes which are time consuming (and limited to those with an OS which can run the required tasks) automated via [Githubs actions](https://github.com/features/actions)?

Look at:
1. NPM release
2. Documentation website release (See https://github.com/netlify/actions)

Note only mocha repository collaborators will have permission create workflows.

**Comment by boneskull** [comment]
release might be problematic. we should be using npm's 2FA, which--last time I checked--means a machine cannot publish.  I'll poke around and see what that looks like.

tbh, I'm interested in moving the entire build to GH actions.  I've been using it in my other projects, and it's nice & fast.  not sure what the SauceLabs integration looks like though.

**Comment by craigtaub** [comment]
> I'll poke around and see what that looks like.

Thanks man. Let's break it down and see where the issues are.

**Comment by outsideris** [comment]
We can use Actions to trigger Netlify( #3644 ) and replace AppVeyor for windows testing ( #4013 )

**Comment by juergba** [comment]
Regarding publishing and 2FA: could [Wombat](https://github.com/GoogleCloudPlatform/wombat-dressing-room) evtl. be a solution?

**Comment by cspotcode** [comment]
Moving to Github Actions means running everything in docker, which conveniently means a consistent environment for running stuff locally, too.  So e.g. folks on Windows can publish as long as they have docker installed.

For publishing, you can tell npm to publish a pre-built tarball
https://docs.npmjs.com/cli/publish
This means Github Actions could do the entire build, version bump, prepare the release notes, push git tags, then post the tarball for a maintainer to publish.  The maintainer would run a (dockerized) publication script on their machine that downloads the tarball from Github and runs `npm publish next-version.tar.gz`

**Comment by juergba** [comment]
@cspotcode Interesting.
On my Windows I have WSL2.0 / Ubuntu / VSCode WSL-extension installed. As an alternative to your docker solution, I just need to install npm on Ubuntu side, then I could run `npm publish downloaded-tarball.tar.gz`. Right?

**Comment by cspotcode** [comment]
Even on Windows without WSL, that should work correctly.

**Comment by ssbarnea** [comment]
How active is mocha development? I see an interesting https://github.com/mochajs/mocha/pull/4822 which is still waiting for the approval to run the tests for 3 days.

**Comment by msimerson** [comment]
> we should be using npm's 2FA, which--last time I checked--means a machine cannot publish.

Yes we should. And you can set up NPM with 2FA and still publish. You have to log into NPM with the publishing account and set up an _Access Token_, which you save in GitHub as a secret. Then you configure a GHA as I've done in [this example](https://github.com/msimerson/node-lts-versions/blob/main/.github/workflows/publish.yml).
